miniapps/meshing/ext-mesh-mapping.cpp

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+//
+//             ---------------------------------------------------
+//                       External Mesh Mapping Miniapp
+//             ---------------------------------------------------
+//
+// This miniapp starts with a serial non-conforming mesh in a dummy format and demonstrates
+// how to build up a corresponding non-conforming MFEM Mesh and then decompose it
+// into a parallel ParMesh.  As part of this process we will demonstrate how to obtain
+// and compose the vertex ID mappings from the various steps that can shuffle the
+// vertices.  In the end this will let us map between the vertex ID numbers in the
+// external dummy mesh and the parallel non-conforming mesh constructed in MFEM.  This
+// is the sort of thing you will have to do if you intend to add MFEM meshes to an existing
+// simulation code and need them to exist and share data with other kinds of meshes in that
+// code.  If you are starting a new MFEM code, it makes much more sense to do everything with
```

Comment:
It could just be me but I have trouble following this sentence which starts with "This is the sort of thing...".  What does it mean to "add MFEM meshes to an existing simulation code"? If I understand correctly the original mesh file is in a non-MFEM format so I don't think you mean an MFEM mesh file. Does this refer to an `mfem::Mesh` object being added to an existing simulation code?

I don't think we need too much detail but a little more clarification of the intended use case would help.

Reply:
Also please trim comments to 80 symbols per line.

Reply:
Thanks for having a look!  I'll fix up the comments and let you know when to have another look.